lib/assets/javascripts/turbograft/response.coffee

```diff
@@ -0,0 +1,26 @@
+class TurboGraft.Response
+  constructor: (@xhr) ->
+
+  valid: -> !@clientOrServerError() && @validContent()
+
+  document: ->
+    if @valid()
+      TurboGraft.Document.create(@xhr.responseText)
+
+  clientOrServerError: ->
+    return false if @xhr.status == 422 # we want to render form validations
+    400 <= @xhr.status < 600
+
+  validContent: ->
+    if contentType = @xhr.getResponseHeader('Content-Type')
+      contentType.match(/^(?:text\/html|application\/xhtml\+xml|application\/xml)(?:;|$)/)
+    else
+      throw @xhrError()
+
+  xhrError: () ->
```

Comment:
only line 18 calls this function, so you could just throw it directly from there?